src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/IllegalIdentifierNameCheck.java

```diff
@@ -27,9 +27,9 @@
 /**
  * <p>
  * Checks identifiers with a pattern for a set of illegal names, such as those
- * that are restricted or contextual keywords. Examples include "yield", "record", "_", and
+ * that are restricted or contextual keywords. Examples include "yield", "record", and
  * "var". Please read more at
- * <a href="https://docs.oracle.com/javase/specs/jls/se14/html/jls-3.html#jls-3.9">
+ * <a href="https://docs.oracle.com/javase/specs/jls/se22/html/jls-3.html#jls-3.9">
```

Comment:
@mahfouz72 @nrmancuso Now that I check the JLS, I have a question and concern.

https://docs.oracle.com/javase/specs/jls/se22/html/jls-3.html#jls-3.9
It is listed as a reserved keyword. Check description says restricted or **contextual** keywords. I assume this is why this won't be a violation with Java 22+?

However, with Java 16 and below,
https://docs.oracle.com/javase/specs/jls/se16/html/jls-3.html#jls-3.9
The JLS says this is a **keyword**. It doesn't say what type, except it is a keyword. My assumption is we should violate everything listed there for Java 16 and below.

Doesn't this change mean we are breaking with old Java versions? We discussed adding a Java version to another check for a similar situation. Shouldn't we do the same here?

Funny enough, starting with Java 8 and below,
https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.9
The JLS does not even list the underscore at all. I assume this also means we shouldn't have a violation here with Java 8 and below.

Reply:
This can be ignored for now, but me and @nrmancuso are talking this over in private.